clients/src/main/java/org/apache/kafka/common/protocol/types/Schema.java

```diff
@@ -83,10 +92,21 @@ public Struct read(ByteBuffer buffer) {
      */
     @Override
     public int sizeOf(Object o) {
+        return sizeOf(o, false);
+    }
+
+    /**
+     * Size of the raw object which excludes default values.
+     */
+    public int sizeOfRaw(Object o) {
+        return sizeOf(o, true);
+    }
+
+    private int sizeOf(Object o, boolean raw) {
         int size = 0;
         Struct r = (Struct) o;
         for (int i = 0; i < fields.length; i++)
-            size += fields[i].type.sizeOf(r.get(fields[i]));
+            size += fields[i].type.sizeOf(raw ? r.getRaw(fields[i]) : r.get(fields[i]));
```

Comment:
Are we just assuming that this doesn't have nested complex types? Wouldn't you need to call the correct method (`sizeofRaw` vs `sizeOf`) here in addition to getting the raw/processed value? See also question below about whether `sizeOfRaw` should be part of `Type` interface.